File: brainbox/palindrome_number.js
function isPalindrome(x) {
    if (x < 0) return false;
    return String(x) === String(x).split("").reverse().join("");
}


function testIsPalindrome() {
    console.assert(isPalindrome(121) === true, "121 should be true");
    console.assert(isPalindrome(0) === true, "0 should be true");
    console.assert(isPalindrome(1221) === true, "1221 should be true");

    console.assert(isPalindrome(123) === false, "123 should be false");
    console.assert(isPalindrome(10) === false, "10 should be false");
    console.assert(isPalindrome(-1) === false, "-1 should be false");

    console.log("All tests passed!");
}

testIsPalindrome();

File: brainbox/palindrome_number.py
def isPalindrome(x):
    if x < 0:
        return False
    return str(x) == str(x)[::-1]

def test_is_palindrome_true():
    assert isPalindrome(121)
    assert isPalindrome(0)
    assert isPalindrome(1221)


def test_is_palindrome_false():
    assert not isPalindrome(123)
    assert not isPalindrome(10)
    assert not isPalindrome(-1)